pageserver/src/layered_repository.rs

```diff
@@ -1105,73 +1135,92 @@ impl LayeredTimeline {
         // requests will block.
         let mut layers = self.layers.lock().unwrap();
 
-        // Walk through each in-memory, and write any dirty data to disk,
-        // as snapshot files.
-        //
-        // We currently write a new snapshot file for every relation
-        // that was modified, if there has been any changes at all.
-        // It would be smarter to only flush out in-memory layers that
-        // have accumulated a fair amount of changes. Note that the
-        // start and end LSNs of snapshot files belonging to different
-        // relations don't have to line up, although currently they do
-        // because of the way this works. So you could have a snapshot
-        // file covering LSN range 100-200 for one relation, and a
-        // snapshot file covering 150-250 for another relation. The
-        // read functions should even cope with snapshot files
-        // covering overlapping ranges for the same relation, although
-        // that situation never arises currently.
-        //
-        // Note: We aggressively freeze and unload all the layer
-        // structs. Even if a layer is actively being used. This
-        // keeps memory usage in check, but is probably too
-        // aggressive. Some kind of LRU policy would be appropriate.
+        // Take the in-memory layer with the oldest WAL record. If it's older
```

Comment:
Wouldn't there be a problem if a layer has the oldest record, and a bunch of very recent ones, so there might be a high chance that we will need to fetch it from disk soon again?

Reply:
Yep, that can happen. If were smart, we would write it to disk but also keep it in memory in that case. But we're not that smart yet :-).

If we just wrote it out, it is still in the OS cache when it's read back, so that helps a little.